controllers/ManagementController.php

```diff
@@ -201,14 +202,14 @@ private function createAccount(CreateAccountForm $form)
         $user->is_validated = false;
         $user->is_active = true;
         $user->role = $form->role;
-        $token_try = 0;
+        $tokenTry = 0;
 
         do {
             $token = $this->createUserToken();
-            $token_try++;
-        } while (!is_null(User::findOne(['token' => $token])) AND $token_try < 10);
+            $tokenTry++;
+        } while (!is_null(User::findOne(['token' => $token])) AND $tokenTry < 10);
 
-        if($token_try == 10) {
+        if($tokenTry == 10 OR !is_null(User::findOne(['token' => $token]))) {
```

Comment:
That's not goot, it should be AND, not OR. I suggest you move this condition inside the do-while (just before calling `$this->createUserToken()`), and remove the `AND $tokenTry < 10` from your while condition.